### ISSUE TITLE
problem executing actions in versions over 1.5.1

### ISSUE THREAD

**Issue by SilverZerOne** [github/local-action#102]
node version 18 or 20
typescript version 4.9.5
@github/local-action >2.0.0

When I try to execute an action un version >=2.0.0

for example local-action run .github\actions\validate-sgon  ../../../src/main/actions/validateSgonAction.ts .env 

node:fs:453 
    return binding.readFileUtf8(path, stringToFlags(options.flag)); 
                   ^ 
 
Error: ENOENT: no such file or directory, open 'C:\Users\x648580\NodeProjects\gln-api-orchestator-manager-action\.github\actions\validate-sgon\package.json' 
    at Object.readFileSync (node:fs:453:20) 
    at isESM (c:\Users\x648580\AppData\Roaming\npm\node_modules\@github\local-action\src\utils\package.ts:11:8) 
    at action (c:\Users\x648580\AppData\Roaming\npm\node_modules\@github\local-action\src\commands\run.ts:95:7) 
    at Command.<anonymous> (c:\Users\x648580\AppData\Roaming\npm\node_modules\@github\local-action\src\command.ts:112:7) { 
  errno: -4058, 
  code: 'ENOENT', 
  syscall: 'open', 
  path: 'C:\\Users\\x648580\\NodeProjects\\gln-api-orchestator-manager-action\\.github\\actions\\validate-sgon\\package.json' 
} 

BUT when i downgrade to @github/local-action 1.5.1 Its WORK!

My code was compiled with ncc :
"build:sgon": "ncc build --source-map -o .github/actions/validate-sgon src/main/actions/validateSgonAction.ts",

if you need more evidence, im really happy to help us!

**Comment by ncalteen** [comment]
Heyo, thanks for the detailed info! I am checking into this a bit now. Do you by chance have a public repo that can be used to reproduce this error?

**Comment by ncalteen** [comment]
I think I found part of the problem. It's based on the structure of the action repository. Originally I built this tool to support the structure in the [TypeScript](https://github.com/actions/typescript-action) and [JavaScript](https://github.com/actions/javascript-action) action templates. They both look like the following:

```plain
.
├── action.yml
├── src
│   ├── index.ts
│   └── main.ts
└── tsconfig.json
```

However it looks like your repo looks more like this...

```plain
.
├── .github
│   └── actions
│       └── validate-sgon
│           └── action.yml
└── src
    └── main
        └── actions
            └── validateSgonAction.ts
```

When `local-action` runs, it starts by looking for the full path to the action. In your case, something like this: `C:\Users\x648580\NodeProjects\gln-api-orchestator-manager-action\.github\actions\validate-sgon`.

Next, it looks for the `package.json` for the source of the action code. Currently, it expects that to be in the same directory as the `action.yml` (not your issue or anything, I simply didn't take this repo structure into account).

I should be able to set this up to instead traverse the directory structure from the entrypoint (`validateSgonAction.ts`) upwards until it finds the `package.json` :)
